src/core/engines/BaseEngine.sol

```diff
@@ -197,93 +228,144 @@ abstract contract BaseEngine is ReentrancyGuard {
-     * @param _data bytes data to decode
+     * @dev Add long token into the account to reduce capital requirement.
+     * @param _subAccount subaccount that will be update in place
      */
-    function _merge(address _subAccount, bytes memory _data) internal {
+    function _addOption(address _subAccount, bytes memory _data) internal virtual {
         // decode parameters
-        (uint256 longTokenId, uint256 shortTokenId, address from, uint64 amount) = abi.decode(
-            _data,
-            (uint256, uint256, address, uint64)
-        );
+        (uint256 tokenId, uint64 amount, address from) = abi.decode(_data, (uint256, uint64, address));
 
         // token being burn must come from caller or the primary account for this subAccount
         if (from != msg.sender && !_isPrimaryAccountFor(from, _subAccount)) revert BM_InvalidFromAddress();
 
-        _verifyMergeTokenIds(longTokenId, shortTokenId);
+        _verifyLongTokenIdToAdd(tokenId);
+
+        // update the state
+        _increaseLongInAccount(_subAccount, tokenId, amount);
+
+        emit OptionTokenAdded(_subAccount, tokenId, amount);
+
+        // transfer the option token in
+        IERC1155(address(optionToken)).safeTransferFrom(from, address(this), tokenId, amount, "");
+    }
+
+    /**
+     * @dev Remove long token from the account to increase capital requirement.
+     * @param _subAccount subaccount that will be update in place
+     */
+    function _removeOption(address _subAccount, bytes memory _data) internal virtual {
+        // decode parameters
+        (uint256 tokenId, uint64 amount, address to) = abi.decode(_data, (uint256, uint64, address));
+
+        // update the state
+        _decreaseLongInAccount(_subAccount, tokenId, amount);
+
+        emit OptionTokenRemoved(_subAccount, tokenId, amount);
+
+        // transfer the option token in
+        IERC1155(address(optionToken)).safeTransferFrom(address(this), to, tokenId, amount, "");
+    }
+
+    /**
+     * @dev Transfers collateral to another account.
+     * @param _subAccount subaccount that will be update in place
+     */
+    function _transferCollateral(address _subAccount, bytes memory _data) internal virtual {
+        // decode parameters
+        (uint80 amount, address to, uint8 collateralId) = abi.decode(_data, (uint80, address, uint8));
 
         // update the account in state
-        _mergeLongIntoSpread(_subAccount, shortTokenId, longTokenId, amount);
+        _removeCollateralFromAccount(_subAccount, collateralId, amount);
+        _addCollateralToAccount(to, collateralId, amount);
 
-        emit OptionTokenMerged(_subAccount, longTokenId, shortTokenId, amount);
+        (address collateral, ) = grappa.assets(collateralId);
 
-        // this line will revert if usre is trying to burn an un-authrized tokenId
-        optionToken.burn(from, longTokenId, amount);
+        emit CollateralTransfered(_subAccount, to, collateral, amount);
```

Comment:
It would make more sense to only log collateralId, just so you don't need to read collateral address just to log it.
(For an off-chain user, this would be redundant information as they already know which collateral has which id through events)